spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceProperties.java

```diff
@@ -522,37 +522,31 @@ public void setProperties(Map<String, String> properties) {
 
 	static class DataSourceBeanCreationException extends BeanCreationException {
 
+		private final EmbeddedDatabaseConnection connection;
+
+		private final Environment environment;
+
+		private final String property;
+
 		DataSourceBeanCreationException(EmbeddedDatabaseConnection connection,
 				Environment environment, String property) {
-			super(getMessage(connection, environment, property));
+
+			super("Cannot auto-configure DataSource. ");
```

Comment:
If you want to pursue this, the message should be provided by the caller, you  can't hard-code it like that.